packages/auth/src/AuthProvider.tsx

```diff
@@ -149,12 +150,15 @@ export const AuthProvider = (props: AuthProviderProps) => {
   const skipFetchCurrentUser = props.skipFetchCurrentUser || false
 
   const [hasRestoredState, setHasRestoredState] = useState(false)
-
-  const [authProviderState, setAuthProviderState] = useState(
-    defaultAuthProviderState
-  )
-
+  const [authProviderState, setAuthProviderState] = useState(defaultState)
   const [rwClient, setRwClient] = useState<AuthClient>()
+  const isMounted = useRef(true)
```

Comment:
We actually already have a hook for this inside the router package. If we agree on these changes I can either shuffle some code around so we can use the same hook in both packages. Or I just copy it over here (I prefer copying it over)